packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -1605,3 +1613,668 @@ class _RestorableScrollOffset extends RestorableValue<double?> {
   @override
   bool get enabled => value != null;
 }
+
+// 2D SCROLLING
+
+// Vertical outer scrollable of 2D scrolling
+class _VerticalOuterDimension extends Scrollable {
+  const _VerticalOuterDimension({
+    super.key,
+    required super.viewportBuilder,
+    required super.axisDirection,
+    required ScrollController controller,
+    super.physics,
+    super.clipBehavior,
+    super.incrementCalculator,
+    super.excludeFromSemantics,
+    super.dragStartBehavior,
+    super.restorationId,
+  }) : assert(axisDirection == AxisDirection.up || axisDirection == AxisDirection.down),
+       super(controller: controller);
+
+  @override
+  _VerticalOuterDimensionState createState() => _VerticalOuterDimensionState();
+}
+
+class _VerticalOuterDimensionState extends ScrollableState {
+  late DiagonalDragBehavior diagonalDragBehavior;
+
+  @override
+  void didChangeDependencies() {
+    diagonalDragBehavior = TwoDimensionalScrollable._diagonalDragBehavior(context)!;
```

Comment:
Why can this not just be passed in and needs to be looked up?

Reply:
Oh oh! Thank you for pointing this out. This part changed so many times. Since I am extending ScrollableState I cannot access new field member from _Vertical* and _Horizontal* - but I can, I just need to cast the widget.